smi-test/main.go

```diff
@@ -0,0 +1,294 @@
+package main
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"testing"
+
+	harness "github.com/kudobuilder/kuttl/pkg/apis/testharness/v1beta1"
+	"github.com/kudobuilder/kuttl/pkg/test"
+	testutils "github.com/kudobuilder/kuttl/pkg/test/utils"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/discovery"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type URLstruct struct {
+	URL     string
+	Method  string
+	Headers map[string]string
+}
+type MetricResponse struct {
+	ReqReceived   []string
+	RespSucceeded []URLstruct
+	RespFailed    []URLstruct
+}
+
+func getCustomTestsForAccess() map[string]test.CustomTest {
+	testHandlers := make(map[string]test.CustomTest)
+	testHandlers["traffic"] = func(
+		t *testing.T,
+		namespace string,
+		clientFn func(forceNew bool) (client.Client, error),
+		DiscoveryClient func() (discovery.DiscoveryInterface, error),
+		Logger testutils.Logger,
```

Comment:
@nveenjain are we using the "github.com/sirupsen/logrus" package here? // @anandrkskd 

@nveenjain I think you were saying that use of kuttl's logger is best for the use case here, is that right?